wdl-ast/tests/validation/conflicting-call-names/source.errors

```diff
@@ -0,0 +1,103 @@
+error: conflicting call name `my_int`
+  ┌─ tests/validation/conflicting-call-names/source.wdl:7:10
+  │
+6 │     Int my_int = 0      # FIRST
+  │         ------ the declaration with the conflicting name is here
+7 │     call my_int         # OK
+  │          ^^^^^^ this conflicts with a declaration of the same name
+  │
+  = fix: add an `as` clause to the call to specify a different name
```

Comment:
I keep getting this namespace stuff wrong, so I'm tagging @adthrasher as a double check.

But I think this validation check needs to be relaxed. This example here: any downstream usage of `my_int` will be unambiguous to the engine (but very confusing to readers).
- If the full identifier is `my_int` we are referring to the `Int`
- If the full identifier is `my_int.*` we are referring to the `call`

There's never an ambiguous situation arising from this "conflict", so I think it needs to be permitted.

Reply:
Agreed, I think this is allowable. So it should be permitted by this validation check (though maybe flagged by a lint rule?).

Reply:
Also `my_int.x` is not an identifier, it's an expression:

```
AccessExprNode
  NameRefNode
    Ident "my_int"
  Dot "."
  Ident "x"
```

An evaluator resolves the name reference to whatever value has that name; in a well-specified language, the `call` statement produces a value of a particular type and shouldn't be treated different from any other declaration.

Reply:
@peterhuene that all makes sense to me. Is there anything resulting from this popping up in Gauntlet?

Reply:
That's an excellent point @peterhuene. We should collect these examples and bring them up with the WDL maintainers. If our behavior matches `miniwdl`'s, there probably will be little resistance to making that behavior explicit in the spec.

Reply:
> Is there anything resulting from this popping up in Gauntlet?

The change didn't require an update to `Gauntlet.toml`, so I don't think anyone is running into this validation error currently.